be/src/vec/functions/function_string.h

```diff
@@ -386,6 +387,194 @@ class FunctionStrcmp : public IFunction {
+
+            const auto* col_str = assert_cast<const ColumnString*>(col.get());
+            chars_list[i] = &col_str->get_chars();
+            offsets_list[i] = &col_str->get_offsets();
+            is_const_args[i] = is_const;
+        }
+
+        auto res = ColumnString::create();
+        auto& res_data = res->get_chars();
+        auto& res_offset = res->get_offsets();
+        res_offset.resize(input_rows_count);
+        const char* partition_type = chars_list[0]->raw_data();
+        // partition type is list|range
+        if (std::strncmp(partition_type, "list", 4) == 0) {
+            int data_size = 0;
+            for (int i = 0; i < input_rows_count; i++) {
+                std::string res_p = "p";
+                for (int j = 1; j < argument_size; j++) {
+                    const auto& current_offsets = *offsets_list[j];
+                    const auto& current_chars = *chars_list[j];
+
+                    auto idx = index_check_const(i, is_const_args[j]);
+                    int size = current_offsets[idx] - current_offsets[idx - 1];
+
+                    const char* tmp =
+                            reinterpret_cast<const char*>(&current_chars[current_offsets[idx - 1]]);
+                    const std::string string_to_unicode(tmp, size);
+                    auto unicode_to_string = _string_to_u16string(string_to_unicode);
+                    res_p += _string_to_unicode(unicode_to_string) +
+                             std::to_string(unicode_to_string.size());
+                }
+                int len = res_p.size();
+                if (len > 50) {
+                    return Status::Error<ErrorCode::INVALID_ARGUMENT>(
+                            "The list partition name cannot exceed 50 characters");
+                }
+                data_size += len;
+                res_data.resize(data_size);
+                memcpy(&res_data[res_offset[i - 1]], res_p.c_str(), len);
+                res_offset[i] = res_offset[i - 1] + len;
+            }
+        } else {
+            // split date with '-' eg. [2022-12-12]
+            if (argument_size != 3) {
+                return Status::Error<ErrorCode::INVALID_ARGUMENT>(
+                        "The partition type is range, the argument of size must be 3");
+            }
+            const char* range_type = chars_list[1]->raw_data();
+            std::string to_split_s(chars_list[2]->raw_data(), chars_list[2]->size());
+            std::vector<std::string> str_v;
+            // split date with different way [2022-12-12 00:00:00] or [2022-12-12]
+            if (to_split_s.size() == input_rows_count * 19) {
+                for (int i = 0; i < to_split_s.size(); i += 19) {
+                    auto vec_res = split(to_split_s.substr(i, 10), "-");
+                    for (const auto& s : vec_res) {
+                        str_v.emplace_back(s);
+                    }
+                }
+            } else {
```

Comment:
I think it's better to use `else if` here and add an unlikely else branch to throw Exception